src/test/java/com/commercetools/sync/cartdiscounts/utils/CartDiscountUpdateActionUtilsTest.java

```diff
@@ -0,0 +1,563 @@
+    }
+
+    @Test
+    public void buildChangeTargetUpdateAction_WithLineItemAndShippingTargetValues_ShouldBuildUpdateAction() {
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        final CartDiscountTarget cartDiscountTarget = ShippingCostTarget.of();
+        when(newCartDiscountDraft.getTarget()).thenReturn(cartDiscountTarget);
+
+        final Optional<UpdateAction<CartDiscount>> changeTargetUpdateAction =
+            buildChangeTargetUpdateAction(CART_DISCOUNT_WITH_RELATIVE_VALUE, newCartDiscountDraft);
+
+        assertThat(changeTargetUpdateAction).contains(ChangeTarget.of(cartDiscountTarget));
+    }
+
+    @Test
+    public void buildChangeIsActiveUpdateAction_WithDifferentValues_ShouldBuildUpdateAction() {
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        when(newCartDiscountDraft.isActive()).thenReturn(true);
+
+        final Optional<UpdateAction<CartDiscount>> changeIsActiveUpdateAction =
+            buildChangeIsActiveUpdateAction(CART_DISCOUNT_WITH_RELATIVE_VALUE, newCartDiscountDraft);
+
+        assertThat(changeIsActiveUpdateAction).contains(ChangeIsActive.of(true));
+    }
+
+    @Test
+    public void buildChangeIsActiveUpdateAction_WithNullValues_ShouldBuildUpdateAction() {
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        when(newCartDiscountDraft.isActive()).thenReturn(null);
+
+        final Optional<UpdateAction<CartDiscount>> changeIsActiveUpdateAction =
+            buildChangeIsActiveUpdateAction(CART_DISCOUNT_WITH_RELATIVE_VALUE, newCartDiscountDraft);
+
+        assertThat(changeIsActiveUpdateAction).contains(ChangeIsActive.of(true));
+    }
+
+    @Test
+    public void buildChangeIsActiveUpdateAction_WithSameValues_ShouldNotBuildUpdateAction() {
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        when(newCartDiscountDraft.isActive()).thenReturn(false);
+
+        final Optional<UpdateAction<CartDiscount>> changeIsActiveUpdateAction =
+            buildChangeIsActiveUpdateAction(CART_DISCOUNT_WITH_RELATIVE_VALUE, newCartDiscountDraft);
+
+        assertThat(changeIsActiveUpdateAction).isNotPresent();
+    }
+
+    @Test
+    public void buildChangeNameUpdateAction_WithDifferentValues_ShouldBuildUpdateAction() {
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        when(newCartDiscountDraft.getName()).thenReturn(LocalizedString.of(Locale.ENGLISH, "newName"));
+
+        final Optional<UpdateAction<CartDiscount>> changeNameUpdateAction =
+            buildChangeNameUpdateAction(CART_DISCOUNT_WITH_RELATIVE_VALUE, newCartDiscountDraft);
+
+        assertThat(changeNameUpdateAction).contains(ChangeName.of(LocalizedString.of(Locale.ENGLISH, "newName")));
+    }
+
+    @Test
+    public void buildChangeNameUpdateAction_WithNullValues_ShouldBuildUpdateAction() {
```

Comment:
I think you miss understood the `WithNullValues` throughout this test. Usually it means that both old and new are `null`. But in these cases it's only then new one which is `null`

Reply:
What about tests for:

both null values (old and new)?
just old null value?